terraform/api_gateway.tf.json.template.py

```diff
@@ -532,16 +570,41 @@ def for_domain(cls, domain):
                     ],
                     'resource_arn': '${aws_wafv2_web_acl.api_gateway.arn}',
                     'logging_filter': {
-                        'default_behavior': 'DROP',
-                        'filter': {
-                            'behavior': 'KEEP',
-                            'requirement': 'MEETS_ALL',
-                            'condition': {
-                                'action_condition': {
-                                    'action': 'BLOCK'
+                        # We use a 'KEEP' default behavior and selectively drop
+                        # logs that we don't need. This implementation gives us
+                        # filters that only 'DROP', avoiding an issue due
+                        # to https://www.github.com/hashicorp/terraform-provider-aws/issues/32665
+                        # which causes TF to deploy the filters in random order,
+                        # potentially breaking the desired effect when some
+                        # filters 'DROP' and others 'KEEP'.
+                        #
+                        'default_behavior': 'KEEP',
+                        'filter': [
+                            {
+                                'behavior': 'DROP',
+                                'requirement': 'MEETS_ALL',
+                                # The condition(s) listed here should cover all
+                                # the WAF rule actions we wish to drop from the
+                                # logs. Multiple action conditions can be added
+                                # to this one filter.
+                                'condition': {
+                                    'action_condition': {
+                                        'action': 'ALLOW'
+                                    }
+                                }
+                            },
+                            {
+                                'behavior': 'DROP',
+                                'requirement': 'MEETS_ALL',
+                                'condition': {
+                                    'label_name_condition': {
+                                        'label_name': 'awswaf:%s:webacl:' % config.aws_account_id +
+                                                      '${aws_wafv2_web_acl.api_gateway.name}:' +
+                                                      config.blocked_v4_ips_term,
+                                    }
```

Comment:
Consider a list comprehension to reduce the duplication, and refine the comments to make it more concise if they are necessary.